contracts/CToken.sol

```diff
@@ -1058,52 +1070,48 @@ contract CToken is CTokenInterface, Exponential, TokenErrorReporter {
             return fail(Error.INVALID_ACCOUNT_PAIR, FailureInfo.LIQUIDATE_SEIZE_LIQUIDATOR_IS_BORROWER);
         }
 
-        MathError mathErr;
-        uint borrowerTokensNew;
-        uint liquidatorTokensNew;
-        uint liquidatorSeizeTokens;
-        uint protocolSeizeTokens;
-        uint protocolSeizeAmount;
-        uint exchangeRateMantissa;
-        uint totalReservesNew;
+        SeizeInternalLocalVars memory vars;
 
         /*
          * We calculate the new borrower and liquidator token balances, failing on underflow/overflow:
          *  borrowerTokensNew = accountTokens[borrower] - seizeTokens
          *  liquidatorTokensNew = accountTokens[liquidator] + seizeTokens
          */
-        (mathErr, borrowerTokensNew) = subUInt(accountTokens[borrower], seizeTokens);
-        if (mathErr != MathError.NO_ERROR) {
-            return failOpaque(Error.MATH_ERROR, FailureInfo.LIQUIDATE_SEIZE_BALANCE_DECREMENT_FAILED, uint(mathErr));
+        (vars.mathErr, vars.borrowerTokensNew) = subUInt(accountTokens[borrower], seizeTokens);
+        if (vars.mathErr != MathError.NO_ERROR) {
+            return failOpaque(Error.MATH_ERROR, FailureInfo.LIQUIDATE_SEIZE_BALANCE_DECREMENT_FAILED, uint(vars.mathErr));
         }
 
-        protocolSeizeTokens = mul_(seizeTokens, Exp({mantissa: protocolSeizeShareMantissa}));
-        liquidatorSeizeTokens = sub_(seizeTokens, protocolSeizeTokens);
+        vars.protocolSeizeTokens = mul_(seizeTokens, Exp({mantissa: protocolSeizeShareMantissa}));
+        vars.liquidatorSeizeTokens = sub_(seizeTokens, vars.protocolSeizeTokens);
 
-        (mathErr, exchangeRateMantissa) = exchangeRateStoredInternal();
-        require(mathErr == MathError.NO_ERROR, "exchange rate math error");
+        (vars.mathErr, vars.exchangeRateMantissa) = exchangeRateStoredInternal();
+        require(vars.mathErr == MathError.NO_ERROR, "exchange rate math error");
 
-        protocolSeizeAmount = mul_ScalarTruncate(Exp({mantissa: exchangeRateMantissa}), protocolSeizeTokens);
+        vars.protocolSeizeAmount = mul_ScalarTruncate(Exp({mantissa: vars.exchangeRateMantissa}), vars.protocolSeizeTokens);
 
-        totalReservesNew = add_(totalReserves, protocolSeizeAmount);
+        vars.totalReservesNew = add_(totalReserves, vars.protocolSeizeAmount);
+        vars.totalSupplyNew = sub_(totalSupply, vars.protocolSeizeTokens);
```

Comment:
the seized tokens added to reserves are no longer part of the CToken supply

Reply:
🎣